Source/Task/AsyncLib.cpp

```diff
@@ -287,7 +287,7 @@ static HRESULT AllocStateNoCompletion(_Inout_ AsyncBlock* asyncBlock, _Inout_ As
 
     if (state->providerData.queue != nullptr)
     {
-        ReferenceAsyncQueue(state->providerData.queue);
+        DuplicateAsyncQueueHandle(state->providerData.queue);
```

Comment:
For correctness you probably want state->providerData.queue = DuplicateAsyncQueueHandle(...) (even though it returns the input param)